### PR TITLE
Add rocketmq showcase

### PR DESCRIPTION
**Pull request by peachisai**

### CODE REVIEW

**Comment by wu-sheng** [comment]
I can see helm chart, is ut ready?
Please update PR title accordingly.

File: deploy/platform/kubernetes/Makefile.in [hunk 1]
```diff
@@ -23,4 +23,4 @@ SAMPLE_SERVICES_NAMESPACE ?= sample-services
 RELEASE ?= demo
 AGENTLESS ?= false
 
-FEATURE_FLAGS ?= java-agent-injector,cluster,elasticsearch,kubernetes-monitor,so11y,vm-monitor,als,event,istiod-monitor,satellite,rover,trace-profiling,mysql-monitor,postgresql-monitor,nginx-monitor,apisix-monitor,grafana,elasticsearch-monitor,rabbitmq-monitor,mongodb-monitor,r3
+FEATURE_FLAGS ?= java-agent-injector,cluster,elasticsearch,kubernetes-monitor,so11y,vm-monitor,als,event,istiod-monitor,satellite,rover,trace-profiling,mysql-monitor,postgresql-monitor,nginx-monitor,apisix-monitor,grafana,elasticsearch-monitor,rabbitmq-monitor,mongodb-monitor,r3,rocketmq-monitor
```
[the remark this answers is not in the document]
Comment (wu-sheng) [reply deploy/platform/kubernetes/Makefile.in:26]: Is our demo using rabbit or active MQ? @wankai123

**Comment by peachisai** [comment]
> I can see helm chart, is ut ready? Please update PR title accordingly.

Done

File: deploy/platform/kubernetes/Makefile.in (hunk 1 above)
[the remark this answers is not in the document]
Comment (peachisai) [reply deploy/platform/kubernetes/Makefile.in:26]: > Is our demo using rabbit or active MQ? @wankai123

Yes, Demo have both

[the remark this answers is not in the document]
Comment (peachisai) [reply deploy/platform/kubernetes/Makefile.in:26]: As RabbitMQ is existing, Establishing a hierarchy between RabbitMQ monitoring and k8s layer is better?

[the remark this answers is not in the document]
Comment (wu-sheng) [reply deploy/platform/kubernetes/Makefile.in:26]: > As RabbitMQ is existing, Establishing a hierarchy between RabbitMQ monitoring layer and k8s layer is better?

Yes, it is. And it should be from another pull request. 

Further, we should consider to build hierarchy structure between virtual MQ to RabbitMQ server monitoring as well.

[the remark this answers is not in the document]
Comment (peachisai) [reply deploy/platform/kubernetes/Makefile.in:26]: > > As RabbitMQ is existing, Establishing a hierarchy between RabbitMQ monitoring layer and k8s layer is better?
> 
> Yes, it is. And it should be from another pull request.
> 
> Further, we should consider to build hierarchy structure between virtual MQ to RabbitMQ server monitoring as well.

I'd like to contribute this PR

[the remark this answers is not in the document]
Comment (wu-sheng) [reply deploy/platform/kubernetes/Makefile.in:26]: Please submit an issue about this.

File: deploy/platform/kubernetes/templates/feature-rocketmq/resource.yaml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            - name: TZ
+              value: "Asia/Shanghai"
+            - name: NAMESRV_ADDR
+              value: "localhost:9876"
+          command: ["sh", "tools.sh", "org.apache.rocketmq.example.quickstart.Consumer"]
+        - name: rocketmq-exporter
+          image: apache/rocketmq-exporter:0.0.2
+          imagePullPolicy: IfNotPresent
+          ports:
+            - containerPort: 5557
```
Comment (wankai123) [inline deploy/platform/kubernetes/templates/feature-rocketmq/resource.yaml:104]:
```suggestion
            - containerPort: 5557
              name: metrics
```

**Comment by wu-sheng** [comment]
@peachisai I think this doesn't work on our end k8s. Which kind of k8s did you test?

**Comment by peachisai** [comment]
> @peachisai I think this doesn't work on our end k8s. Which kind of k8s did you test?

I use minikube, Due to the resource limiation, I test only rocketmq, it can suceess start

**Comment by peachisai** [comment]
> > @peachisai I think this doesn't work on our end k8s. Which kind of k8s did you test?
> 
> I use minikube, Due to the resource limiation, I test only rocketmq, it can suceess start.
But I am not very familiar with k8s , I just learn it a few day , maybe something is not correct,please point it

**Comment by wu-sheng** [comment]
They can be booted, but it doesn't work as expected. Kai has pointed out, and he is trying to get more details.

File: deploy/platform/kubernetes/templates/feature-rocketmq/resource.yaml (hunk 2 above)
Comment (wankai123) [reply deploy/platform/kubernetes/templates/feature-rocketmq/resource.yaml:104]: the name is required for OTEL scrape

File: deploy/platform/kubernetes/templates/feature-rocketmq/resource.yaml [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  selector:
+    app: rocketmq
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: rocketmq-exporter
+spec:
+  type: NodePort
```
[the remark this answers is not in the document]
Comment (peachisai) [reply deploy/platform/kubernetes/templates/feature-rocketmq/resource.yaml:39]: Sorry ,this is my test port config ,I forgot to restore it

File: deploy/platform/kubernetes/templates/feature-rocketmq/resource.yaml [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+# specific language governing permissions and limitations
+# under the License.
+#
+
+{{- if .Values.features.rocketmqMonitor.enabled }}
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: rocketmq-service
```
Comment (wankai123) [inline deploy/platform/kubernetes/templates/feature-rocketmq/resource.yaml:24]:
```suggestion
  name: rocketmq
```

Comment (wankai123) [reply deploy/platform/kubernetes/templates/feature-rocketmq/resource.yaml:24]: Due to your rocketMQ cluster name is  `rocketmq.{{ .Release.Namespace }}`,  this name will impact the hierarchy relationship with k8s-service

**Review by wankai123 — APPROVED** [review]
Now this PR should be good

**Comment by peachisai** [comment]
@wankai123 
Hi, It seems that showcase page doesn't show RocketMQ, Is anything that I code wrong?

**Comment by wu-sheng** [comment]
Because website is updated at 1800(UTC+8), you will see that in the evening.

**Comment by wankai123** [comment]
> @wankai123 Hi, It seems that showcase page doesn't show RocketMQ, Is anything that I code wrong?

I updated the demo and we can see it now.

**Comment by peachisai** [comment]
> > @wankai123 Hi, It seems that showcase page doesn't show RocketMQ, Is anything that I code wrong?
> 
> I updated the demo and we can see it now.

Dashboard looks invisible

**Comment by wu-sheng** [comment]
showcase seems don't have producer and consumer? Could you add them? Otherwise, it is an empty service only.

**Comment by wu-sheng** [comment]
Or there are some bugs to make the data not showing up. http://demo.skywalking.apache.org/dashboard/ROCKETMQ/Service/cm9ja2V0bXE6OnJvY2tldG1xLnNreXdhbGtpbmctc2hvd2Nhc2U=.1/RocketMQ-Cluster

**Comment by peachisai** [comment]
> Or there are some bugs to make the data not showing up. http://demo.skywalking.apache.org/dashboard/ROCKETMQ/Service/cm9ja2V0bXE6OnJvY2tldG1xLnNreXdhbGtpbmctc2hvd2Nhc2U=.1/RocketMQ-Cluster

Ok, I found It is that my local cache had trigged
image

**Comment by peachisai** [comment]
And I saw the hierarchy had been built
image

**Pull request merged**